packages/xdl/src/detach/AndroidShellApp.js

```diff
@@ -649,6 +690,42 @@ export async function runShellAppModificationsAsync(context, sdkVersion, buildMo
       )
     );
   }
+  if (parseSdkMajorVersion(sdkVersion) >= 39 && updatesCheckAutomaticallyDisabled) {
```

Comment:
This seems like an expensive operation to run often

Reply:
OK, I added a rough memoization for the majorSdkVersion value in the `runShellAppModificationsAsync` method (assuming that's the operation you're talking about here).